pyth_observer/events.py

```diff
@@ -185,23 +185,32 @@ class PriceDeviation(PriceValidationEvent):
 
     def is_valid(self) -> bool:
         delta = self.publisher_aggregate.price - self.price.aggregate.price
-        self.deviation = abs(delta / self.price.aggregate.price) * 100
-
-        if (self.price.is_publishing(self.publisher_key) and
-                self.price.is_aggregate_publishing() and
-                self.deviation > self.threshold):
+        if self.price.aggregate.price == 0:
             return False
+        else:
+            self.deviation = abs(delta / self.price.aggregate.price) * 100
+
+            if (self.price.is_publishing(self.publisher_key) and
+                    self.price.is_aggregate_publishing() and
+                    self.deviation > self.threshold):
+                return False
         return True
 
     def get_event_details(self) -> Tuple[str, List[str]]:
         agg = self.price.aggregate
         published = self.publisher_aggregate
-
-        title = f"{self.publisher_name.upper()} price is {self.deviation:.0f}% off on {self.symbol}"
-        details = [
-            f"Aggregate: {agg.price:.2f} ± {agg.confidence_interval:.2f} (slot {agg.slot})",
-            f"Published:  {published.price:.2f} ± {published.confidence_interval:.2f} (slot {published.slot})",
-        ]
+        if self.price.aggregate.price == 0:
+            title = f"Aggregate price is 0 on {self.symbol}"
+            details = [
+                f"Aggregate: {agg.price:.2f} ± {agg.confidence_interval:.2f} (slot {agg.slot})",
+                f"Published:  {published.price:.2f} ± {published.confidence_interval:.2f} (slot {published.slot})",
+            ]
```

Comment:
This case won't ever trigger. The event details are only printed if is_valid == True. I think it makes sense to separate out the validation of the aggregate price into a separate PriceValidationEvent though, so no need for this case here.